node/podcontroller.go

```diff
@@ -270,13 +270,19 @@ func (pc *PodController) Run(ctx context.Context, podSyncWorkers int) (retErr er
 		},
 		UpdateFunc: func(oldObj, newObj interface{}) {
 			// Create a copy of the old and new pod objects so we don't mutate the cache.
+			oldPod := oldObj.(*corev1.Pod)
 			newPod := newObj.(*corev1.Pod)
 
 			// At this point we know that something in .metadata or .spec has changed, so we must proceed to sync the pod.
 			if key, err := cache.MetaNamespaceKeyFunc(newPod); err != nil {
 				log.G(ctx).Error(err)
 			} else {
-				pc.k8sQ.AddRateLimited(key)
+				// ignore ResourceVersion change
+				oldPod.ResourceVersion = newPod.ResourceVersion
+				if !cmp.Equal(oldPod.Spec, newPod.Spec) ||
```

Comment:
If you're trying to achieve update suppression, you can use: https://github.com/virtual-kubelet/virtual-kubelet/blob/a67cfab42bc77a441636801d906efceba0683da7/node/pod.go#L103

Also, you need to check deletion timestamp and phase change IIRC

Reply:
No, wo should  Ignore status update here. But we need check `spec`, `deleteTime`, `Annotation`, `Labels` and so on. So I  exclude pod status comparison

Reply:
Please add a more targeted equal method that tests the super set of the attributes of podequal. Something like:
```
podNeedsWork(old, new *v1.pod) bool {
 if !podsEqual(...) { return true };
 if ... return true
 return false
}
```

I'm curious, where is the perf concern?

Reply:
Done, thanks, PTAL thanks